[qa] Docker build fails: missing tsconfig.json in build context

## Bug Description

The Dockerfile introduced in #223 fails to build because it doesn't copy the root `tsconfig.json` into the build context.

## Reproduction

```bash
podman build -t opencara-test .
# or: docker build -t opencara-test .
```

## Error

```
packages/shared build: error TS5083: Cannot read file '/app/tsconfig.json'.
packages/shared build: src/review-config.ts(1,36): error TS2307: Cannot find module 'yaml' or its corresponding type declarations.
```

## Root Cause

`packages/shared/tsconfig.json` extends `../../tsconfig.json` (the root tsconfig). The Dockerfile copies package source files but not the root `tsconfig.json`, so the TypeScript build fails.

## Fix

Add `COPY tsconfig.json ./` to the Dockerfile before the build step (after copying package files):

```dockerfile
COPY pnpm-lock.yaml pnpm-workspace.yaml package.json ./
COPY tsconfig.json ./
COPY packages/shared/package.json packages/shared/
COPY packages/cli/package.json packages/cli/
RUN pnpm install --frozen-lockfile
COPY packages/shared/ packages/shared/
COPY packages/cli/ packages/cli/
RUN pnpm --filter opencara... build
```

## Context

Found during QA verification of PRs #206-#224.

--- a/Dockerfile
+++ b/Dockerfile
@@ -3,6 +3,7 @@ FROM node:20-slim AS builder
 WORKDIR /app
 RUN corepack enable
 COPY pnpm-lock.yaml pnpm-workspace.yaml package.json ./
+COPY tsconfig.json ./
 COPY packages/shared/package.json packages/shared/
 COPY packages/cli/package.json packages/cli/
 RUN pnpm install --frozen-lockfile

